l3build/l3build.lua

```diff
@@ -2069,12 +2070,12 @@ end
 
 -- Unpack the package files using an 'isolated' system: this requires
 -- a copy of the 'basic' DocStrip program, which is used then removed
-function unpack()
+function unpack(sourcedir)
```

Comment:
The parameter is solemnly used to pass on to `bundleunpack`. For both, it is optional. Nonetheless, I agree that user-facing functions should indeed have easy understandable parameters.

For `unpack`, a user may expect a parameter to specify a file to unpack from or a file to generate. (The latter is difficult already as it is unknown in advance.) So I agree that "directory to collect files for unpacking" is a rather unintuitive parameter.

The parameter could be hidden by moving all `unpack` code to an internal function, turning the user-facing `unpack` into a simple wrapper.

Alternatively, it could be avoided completely by writing a dedicated initialization function for the `doc` function, similar to how `check` already has `checkinit`. On the other hand, this would largely duplicate code from the `unpack` and `checkinit` functions. Would this be preferable?